src/services/entities.ts

```diff
@@ -690,70 +791,73 @@ function createDealAcceptedEntity<T>(event: T): void {
         poolCreatedEntity.sponsor.toHex() + '-' + event.address.toHex(),
       )
 
       vestingDealEntity.poolName = poolCreatedEntity.name
       vestingDealEntity.poolAddress = event.address
-      vestingDealEntity.tokenToVest = dealEntity.underlyingDealToken!
-      vestingDealEntity.tokenToVestSymbol = dealEntity.underlyingDealTokenSymbol!
-      vestingDealEntity.investorDealTotal = investorDealTotal
+      vestingDealEntity.tokenToVest = upFrontDealEntity.underlyingDealToken
+      vestingDealEntity.tokenToVestSymbol = upFrontDealEntity.underlyingDealTokenSymbol
+      vestingDealEntity.investorDealTotal = event.params.amountMinted
       vestingDealEntity.remainingAmountToVest = vestingDealEntity.investorDealTotal!
       vestingDealEntity.totalVested = BigInt.fromI32(0)
-      vestingDealEntity.vestingPeriodStarts = dealEntity.vestingPeriodStarts!
-      vestingDealEntity.vestingPeriodEnds = dealEntity.vestingPeriodStarts!.plus(
-        dealEntity.vestingPeriod!,
-      )
-      vestingDealEntity.underlyingDealTokenDecimals = dealEntity.underlyingDealTokenDecimals
+      vestingDealEntity.vestingPeriodStarts = poolCreatedEntity.vestingStarts!
+      vestingDealEntity.vestingPeriodEnds = poolCreatedEntity.vestingEnds!
+      vestingDealEntity.underlyingDealTokenDecimals = upFrontDealEntity.underlyingDealTokenDecimals
       vestingDealEntity.user = poolCreatedEntity.sponsor.toHex()
       vestingDealEntity.pool = poolCreatedEntity.id
     } else {
-      vestingDealEntity.investorDealTotal =
-        vestingDealEntity.investorDealTotal!.plus(investorDealTotal)
+      vestingDealEntity.investorDealTotal = vestingDealEntity.investorDealTotal!.plus(
+        event.params.amountMinted,
+      )
       vestingDealEntity.remainingAmountToVest = vestingDealEntity.investorDealTotal!
     }
 
     vestingDealEntity.save()
   }
 }
 
-export function createOrUpdateSponsorVestingDeal(event: AcceptDealEvent): void {
+export function createOrUpdateSponsorVestingDeal(event: PoolSponsorClaimEvent): void {
   let poolCreatedEntity = getPoolCreated(event.address.toHex())
-  let dealEntity = getDeal(event.params.dealAddress.toHex())
 
-  if (dealEntity === null || poolCreatedEntity === null) {
+  if (poolCreatedEntity === null) {
+    return
+  }
+
+  let dealEntity = getDeal(poolCreatedEntity.dealAddress!.toHex())
+
+  if (dealEntity === null) {
     return
   }
 
   let vestingDealEntity = getVestingDeal(
-    poolCreatedEntity.sponsor.toHex() + '-' + event.params.dealAddress.toHex(),
+    poolCreatedEntity.sponsor.toHex() + '-' + poolCreatedEntity.dealAddress!.toHex(),
   )
 
-  let aelinDealContract = AelinDealContract.bind(event.params.dealAddress)
+  let aelinDealContract = AelinDealContract.bind(Address.fromBytes(poolCreatedEntity.dealAddress!))
   let underlyingPerDealExchangeRate = aelinDealContract.underlyingPerDealExchangeRate()
-  let investorDealTotal = event.params.sponsorFee
+
+  let investorDealTotal = event.params.amountMinted
```

Comment:
And here should be `sponsorFeeAmount` param. `amountMinted` is from upfront deal `SponsorClaim` event